src/pre_process/m_perturbation.fpp

```diff
@@ -0,0 +1,616 @@
+!>
+!! @file m_perturbation.fpp
+!! @brief Contains module m_perturbation
+
+!> @brief This module contains subroutines that compute perturbations to the
+!!              initial mean flow fields.
+module m_perturbation
+
+    ! Dependencies =============================================================
+    use m_derived_types         ! Definitions of the derived types
+
+    use m_global_parameters     ! Global parameters for the code
+
+    use m_mpi_proxy              !< Message passing interface (MPI) module proxy
+
+    use m_eigen_solver          ! Subroutines to solve eigenvalue problem for
+    ! complex general matrix
+
+    use ieee_arithmetic
+
+    ! ==========================================================================
+    ! ==========================================================================
+
+    implicit none
+
+    integer :: mixlayer_nvar ! Number of variables in linear stability analysis solver for mixing layer
+    integer, allocatable, dimension(:) :: mixlayer_var ! Index of variables in linear stability analysis solver
+
+contains
+
+    subroutine s_initialize_perturbation_module()
+
+        if (mixlayer_perturb) then
+            if (model_eqns == 2 .and. num_fluids == 1) then
+                mixlayer_nvar = 5 ! 1 continuity + 3 momentum + 1 energy
+                allocate (mixlayer_var(mixlayer_nvar))
+
+                mixlayer_var(1) = contxb
+                mixlayer_var(2) = momxb
+                mixlayer_var(3) = momxb + 1
+                mixlayer_var(4) = momxb + 2
+                mixlayer_var(5) = momxb + 3
+            end if
+        end if
+
+    end subroutine s_initialize_perturbation_module
+
+    subroutine s_perturb_sphere(q_prim_vf)
```

Comment:
you added this subroutine but it looks like there isn't a test for it

Reply:
`s_perturb_sphere` and `s_perturb_surrounding_flow` are not written by myself. I guess they were written many years ago. I just moved these subroutines to `m_perturbation.fpp` along with instability wave subroutines because it seems more appropriate to put the "perturbation" stuff together in a separate module. I don't think any of the example cases use these subroutines. Maybe these subroutines can be removed then?

Reply:
ah i see, nevermind!